core-web/libs/template-builder/src/lib/components/template-builder/template-builder.component.ts

```diff
@@ -191,4 +198,58 @@ export class TemplateBuilderComponent implements OnInit, AfterViewInit, OnDestro
     deleteRow(id: numberOrString): void {
         this.store.removeRow(id as string);
     }
+
+    /**
+     * @description This method opens the dialog to edit the row styleclasses
+     *
+     * @param {numberOrString} rowID
+     * @memberof TemplateBuilderComponent
+     */
+    editRowStyleClasses(rowID: numberOrString, styleClasses: string[]): void {
+        this.ref = this.dialogService.open(AddStyleClassesDialogComponent, {
+            header: 'Edit Classes',
+            data: {
+                selectedClasses: styleClasses ?? []
+            },
+            resizable: false
+        });
+
+        this.ref.onClose
+            .pipe(
+                take(1),
+                filter((styleClasses) => styleClasses)
+            )
+            .subscribe((styleClasses: string[]) => {
+                this.store.updateRow({ id: rowID as string, styleClass: styleClasses });
+            });
```

Comment:
This code is duplicated in `editRowStyleClasses` and `editBoxStyleClasses `, we can create a function for this.

```suggestion
        this.openDynamicDialog(stylesClasses)
            .subscribe((styleClasses: string[]) => {
                this.store.updateRow({ id: rowID as string, styleClass: styleClasses });
            });
```

And the `openDynamicDialog` can be defined:

```ts
openDynamicDialog(selectedClasses = []): Observable {
      this.ref = this.dialogService.open(AddStyleClassesDialogComponent, {
              header: 'Edit Classes',
              data: {
                  selectedClasses: styleClasses ?? []
              },
              resizable: false
          });

        return this.ref.onClose
            .pipe(
                take(1),
                filter((styleClasses) => styleClasses)
            )
}
```